src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.NoCommand;
+import duke.task.TaskType;
+
+/**
+ * The Parser class will deal with making sense of the user command
+ */
+public class Parser {
+    /**
+     * Reads the input to determine which type of command Duke
+     * should execute to reply the user.
+     *
+     * @param input The user input entered in the command line.
+     * @return The Command to be executed.
+     * @throws DukeException if the input is formatted wrongly.
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] inputArray = input.split(" ", 2);
+        String inputCommand = inputArray[0];
+        String inputDetails;
+        String[] taskDetails;
+        String taskName, taskTime;
+        switch (inputCommand) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "done":
+            if (inputArray.length == 1 || inputArray[1].trim().equals("")) {
```

Comment:
Would it be better to extract out the logic for case "done" into a function to make the switch case statements look cleaner?